src/app/services/compute.service.ts

```diff
@@ -133,8 +133,8 @@ public number;
             flatMap((subscription) => {
                 const locationsUri = `subscriptions/${subscription.subscriptionId}/locations`;
                 return this.azure.get<ArmListResponse<ArmLocationAttributes>>(subscription, locationsUri).pipe(
-                    flatMap(response => response.value.map(x => new ArmLocation(x))),
-                    flatMap(armLocation => {
+                    map(response => response.value.map(x => new ArmLocation(x))),
+                    map(armLocations => armLocations.map(armLocation => {
```

Comment:
this one is good to be `flatMap` or `switchMap`. I think its better to use those than map then mergeAll or switchAll or concatAll

Basically: 
- map: Just take each value emitted by an observable and return a modified value
- flatMap: Take each value emitted by an observable and return another observable(With 1 or multiple values)
- switchMap: Same as flatMap but will cancel if you unsubscribe.

Reply:
Understood, makes sense. Interestingly if we have a `map`  instead of `flatmap` for 
```
map(response => response.value.map(x => new ArmLocation(x))),
```
it requires concatAll to get from an Observable<Observable<Resource[]>> to just a single Observable (due to the inner map for `ArmLocations.map()`).

Can get rid of the mergeAll though.

Reply:
oh is value an observable?

Reply:
ok, I looked at it a bit more. so the first map should stay a `map` the 2nd can be a `switchMap` or `flatMap`

Reply:
also I would move the3 content of that 2nd map into a separate function as its pretty big on its own